nipoppy_cli/nipoppy/data/examples/sample_invocations/heudiconv-0.12.2-prepare.json

```diff
@@ -1,3 +1,9 @@
 {
+    "dicom_dir_template": "[[NIPOPPY_DPATH_SOURCEDATA]]/[[NIPOPPY_PARTICIPANT_ID]]/[[NIPOPPY_BIDS_SESSION]]/*",
+    "subjs": "[[NIPOPPY_PARTICIPANT_ID]]",
+    "converter": "none",
```

Comment:
why put this `converter` flag in invocation and not in descriptor like for the `convert` step? 
Same for `heuristic` flag below.

Reply:
The `convert` step uses the default `converter` (`dcm2niix`), which is already in the descriptor by default. For `prepare` we explicitly set it to `none` so that it doesn't do the conversion. I can add it for the default if you think it would be better but I don't think users would want to change it since the only two options are `dcm2niix` (do the conversion) or `none` (don't do the conversion).

For the `heuristic` flag, it is optional and not used in the `prepare` step so shouldn't be specified then. And usually at that stage in the process we don't even have a heuristic file.

Reply:
If we don't want users to change anything mistakenly and has only 1 possible option per step - wouldn't it better to keep these two in descriptor that maps 1-1 onto the step?

Reply:
In general I think it would be better to have it in invocation files (not the descriptor) for better visibility (same reasoning as your comment about the output spaces in fMRIPrep).

If people are looking at the invocations then we assume they sort of know what they are doing, so if they want to change the default invocation we provide then I say it's up to them. 

I can add `"converter": "dcm2niix"` to the invocation for the `convert` step (again, similar to `output_spaces` in fMRIPrep). But if we set it directly in the descriptor it would lose visibility, and I would rather take the chance that users mess up their invocations than users not knowing that we are changing the default behaviour of the software through the descriptor.

For `heuristic`, hardcoding it in the descriptor for `convert` is doable (it's what we had before this PR), but the whole point of this PR was that we wanted to move the Nipoppy-specific default values outside of the descriptor and inside the invocation...